### PR TITLE
Convert workspaces with id MaskWorkspace

### PR DESCRIPTION
**Pull request by OwenArnold**
Convert mantids MaskWorkspace to a scipp DataArray.

Output of `DataArray` seems like the right approach, however we technically have no "data". For this reason I create a new mask, but duplicate that also as the data value.

### CODE REVIEW

File: python/src/scipp/compat/mantid.py [hunk 1]
```diff
@@ -481,19 +481,29 @@ def convert_monitors_ws(ws, converter, **ignored):
...
-    coords_labs_data["data"] = sc.Variable([spec_dim, dim],
-                                           unit=data_unit,
-                                           values=ws.extractY(),
-                                           variances=stddev2)
+    if as_mask_workspace:
+        coords_labs_data["data"] = sc.Variable([spec_dim, dim],
+                                               unit=data_unit,
+                                               values=ws.extractY(),
+                                               dtype=sc.dtype.bool)
+        coords_labs_data["masks"] = {'mask': coords_labs_data["data"].copy()}
```
Comment (SimonHeybrock) [inline python/src/scipp/compat/mantid.py:499]: I don't think it makes sense to have this as mask and data. Just make it the data, users can then do `data.masks['mask'] = mask.data`.

File: python/src/scipp/compat/mantid.py [hunk 2]
```diff
@@ -688,17 +700,17 @@ def from_mantid(workspace, **kwargs):
             workspace, monitor_ws = mantid.ExtractMonitors(workspace)
             workspaces_to_delete.append(workspace)
             workspaces_to_delete.append(monitor_ws)
-        scipp_obj = convert_Workspace2D_to_data_array(workspace, **kwargs)
-    elif workspace.id() == 'EventWorkspace':
+        scipp_obj = convert_Workspace2D_to_data_array(
+            workspace, as_mask_workspace=is_mask_ws, **kwargs)
```
Comment (SimonHeybrock) [inline python/src/scipp/compat/mantid.py:704]: Wouldn't it be nicer to check the ID in `convert_Workspace2D_to_data_array` so all the flag handling can be removed?

File: python/tests/compat/test_mantid.py [hunk 3]
```diff
@@ -615,5 +615,15 @@ def test_to_workspace_2d_handles_single_x_array():
...
 
 
+def test_from_mask_workspace():
+    from mantid.simpleapi import LoadMask
+    mask = LoadMask('HYS', 'HYS_mask.xml')
+    da = sc.compat.mantid.from_mantid(mask)
+    assert 'mask' in da.masks
+    assert da.data == da.masks['mask']
+    assert da.data.dtype == sc.dtype.bool
+    assert da.dims == ['spectrum', 'empty']
```
Comment (SimonHeybrock) [inline python/tests/compat/test_mantid.py:625]:
```suggestion
    assert da.dims == ['spectrum', 'empty']
    assert da.variances is None
```

**Comment by SimonHeybrock** [comment]
@OwenArnold You have some build fails.

File: python/tests/compat/test_mantid.py [hunk 4]
```diff
@@ -615,5 +615,18 @@ def test_to_workspace_2d_handles_single_x_array():
...
+@pytest.mark.skipif(not mantid_is_available(),
+                    reason='Mantid framework is unavailable')
+def test_from_mask_workspace():
+    from mantid.simpleapi import LoadMask
+    from os import path
+    dir_path = path.dirname(path.realpath(__file__))
+    mask = LoadMask('HYS', path.join(dir_path, 'HYS_mask.xml'))
+    da = sc.compat.mantid.from_mantid(mask)
+    assert da.data.dtype == sc.dtype.bool
+    assert da.dims == ['spectrum', 'empty']
```
Comment (SimonHeybrock) [inline python/tests/compat/test_mantid.py:627]: Actually I think we should just have `spectrum` here?

Comment (OwenArnold) [reply python/tests/compat/test_mantid.py:627]: Yes. SpecialWorkspace types are spectrum-only I believe. Will change.

**Pull request merged**